### ISSUE TITLE
version_utils: incorrect handling of epoch in _rpm_re regex

### ISSUE THREAD

**Issue by thoger** [ihiji/version_utils#7]
The `_rpm_re` regular expression is used to split RPM nvr or nevr to name, epoch, version and release.  However, the regex is incorrect and likely to treat part of the version number as epoch number, as it only handles `:` as optional, rather than whole `(\d*):`.

```
>>> from version_utils import rpm
>>> pkg = rpm.package('firefox-45.0-4.fc22.x86_64')
>>> print pkg.epoch, pkg.version
4 5.0
```

while the expected output is:

```
0 45.0
```

I think the relevant part of the regex should be `(?:(\d*):)?`.  Non-capturing group is used to avoid further changes to `parse_package()`.  However, the following still needs fixing:

```
epoch = default_epoch if epoch == '' else epoch
```

to check `if epoch is None`.

**Comment by mplanchard** [comment]
Hi, and thanks so much for your contribution! You're absolutely right. My list of test packages didn't expose this issue, so I'll implement the suggested changes and add your example package to the test suite.

I hope to have an updated patch version out by the end of the day.

**Comment by mplanchard** [comment]
@thoger I've released version 0.2.3 - please let me know if this does not fix the issue, and thanks again for your contribution.
